.github/workflows/goreleaser.yml

```diff
@@ -1,32 +1,38 @@
 name: goreleaser
 
 permissions:
-  # github releases
   contents: write
 
 on:
   push:
     tags:
       - "v*.*.*"
+  # Validate on develop, main, and master branches that the releaser
+  # is working as expected.
+  pull_request:
+    branches:
+      - develop
+      - main
+      - master
+
 jobs:
   goreleaser:
     runs-on: ubuntu-latest
-    environment: release
     steps:
-      - uses: actions/checkout@v4
+      - name: Checkout code
+        uses: actions/checkout@v4
         with:
           submodules: true
       - name: Set up Go
         uses: actions/setup-go@v4
         with:
           go-version: '1.21.11'
           check-latest: true
-      - name: release dry run
+      - name: Release dry run
```

Comment:
Please explain what you mean by `not fully configured` ?

Reply:
Suggest a fix listing what is missing. I don't see any problems.

Reply:
All of this is done in the goreleaser config and CLI. It validates the config, runs it, and even has pre-release hooks. It does not do any deployment, since this is a dry run.